### PR TITLE
Fix reports view returns a 500

### PR DESCRIPTION
**Pull request by noliveleger**
## Checklist

1. [x] Ensure the tests pass
2. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
3. [x] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)

## Description

Reports are broken in public collections branch.

### CODE REVIEW

File: kpi/models/object_permission.py [hunk 1]
```diff
@@ -185,8 +185,9 @@ def get_cached_code_names(model_: models.Model = None) -> dict:
 
 
 @cache_for_request
-def get_perm_ids_from_code_names(code_names: Union[str, list],
-                                 model_: models.Model = None) -> Union[int, list]:
+def get_perm_ids_from_code_names(
+    code_names: Union[str, list, tuple, set], model_: models.Model = None
+) -> Union[int, list]:
```
Comment (jnm) [inline kpi/models/object_permission.py:190]: Can we use `Collection`? https://stackoverflow.com/a/61155469/2402324

Comment (jnm) [reply kpi/models/object_permission.py:190]: OK, no:
```
In [1]: from collections.abc import Collection                                                                                       

In [2]: _list = list('hi there')                                                                                                     

In [3]: _tup = tuple(_list)                                                                                                          

In [4]: _set = set(_list)                                                                                                            

In [5]: isinstance(_list, Collection)                                                                                                
Out[5]: True

In [6]: isinstance(_tup, Collection)                                                                                                 
Out[6]: True

In [7]: isinstance(_set, Collection)                                                                                                 
Out[7]: True

In [8]: isinstance('ouch', Collection)                                                                                               
Out[8]: True
```

File: kobo/apps/reports/views.py [hunk 2]
```diff
@@ -66,14 +66,14 @@ def get_queryset(self):
...
+                parent__userassetsubscription__user=self.request.user
             ),
             all_perms_required=False,
         )
 
         # Find which of these are deployed, using a custom manager method
         deployed_assets = (
             owned_and_explicitly_shared | subscribed_and_public
-        ) & Asset.objects.deployed()
+        ) & Asset.objects.deployed().distinct()
```
Comment (jnm) [inline kobo/apps/reports/views.py:77]: :snail: :question:

**Review by jnm — COMMENTED** [review]
Discussing in Zulip

File: kobo/apps/reports/views.py (hunk 2 above)
Comment (jnm) [reply kobo/apps/reports/views.py:77]: > Django complains  if joining two queries with one without `distinct()`

File: kpi/models/object_permission.py (hunk 1 above)
Comment (noliveleger) [reply kpi/models/object_permission.py:190]: I tried:. Also tried (Iterable, Sequence), no luck.

**Pull request merged**